tests/unit/moduleapi/testrdb.tcl

```diff
@@ -132,102 +132,121 @@ tags "modules" {
                         # Wait for the replica to start reading the rdb
                         wait_for_condition 100 100 {
                             [s -1 async_loading] eq 1
                         } else {
                             fail "Replica didn't get into async_loading mode"
                         }
+                    }
+
+                    switch $testType {
+                        "Aborted" {
+                            # Make sure that next sync will not start immediately so that we can catch the replica in between syncs
+                            $master config set repl-diskless-sync-delay 5
 
-                        if {$state == "ABORTED"} {
                             # Kill the replica connection on the master
                             set killed [$master client kill type replica]
-                        }
-
-                        # Speed things up
-                        $master config set rdb-key-save-delay 0
 
-                        # Wait for loading to stop (when ABORTED, stop fast)
-                        wait_for_condition 100 100 {
-                            [s -1 async_loading] eq 0
-                        } else {
-                            if {$state == "ABORTED"}
+                            # Wait for loading to stop (fail)
+                            wait_for_condition 100 100 {
+                                [s -1 async_loading] eq 0
+                            } else {
                                 fail "Replica didn't disconnect"
-                            else
-                                fail "Loading didn't stop"
+                            }
+                            
+                            test {Diskless load swapdb RedisModuleEvent_ReplAsyncLoad handling: when loading aborted, can keep module variable same as before} {
+                                assert_equal [$replica dbsize] 2000
+                                assert_equal value1_replica [$replica testrdb.get.before]
+                            }
                         }
-
-                        if {$state == "ABORTED"} {
-                            # Module variable is same as before
-                            assert_equal [$replica dbsize] 2000
-                            assert_equal value1_replica [$replica testrdb.get.before]
-                        } elseif {$state == "COMPLETED"} {
-                            # Module variable got value from master
-                            assert_equal [$replica dbsize] 510
-                            assert_equal value1_master [$replica testrdb.get.before]
+                        "Delayed" {
+                            test {Diskless load swapdb RedisModuleEvent_ReplAsyncLoad handling: during loading, can keep module variable same as before} {
+                                assert_equal [$replica dbsize] 2000
+                                assert_equal value1_replica [$replica testrdb.get.before]
+                            }
```

Comment:
same here... we can run that same check (the value of `testrdb.get.before`), both before we abort and after.

one more interesting complication, maybe we can find a way to make sure we abort only after the module got the callback and loaded a new value into it's temp variable?
i.e. without that check, there's a race, and maybe in some cases we abort even before...
maybe we'll add a command like `testrdb.async_loading.get.before` and do a `wait_for_condition` on it?
i know it means that module is no longer a naive module that uses the new api to get things done, but it does improve the test...

WDYT?

Reply:
Good point about the race. Suggestions applied.
Thanks